### PR TITLE
[4.0] Users and User groups: Separate columns for group and permissions button

### PR DESCRIPTION
**Pull request by chmst**
### Summary of Changes
In user groups list view: Bring the "advanced permission" link to an own column.

usergroups

users

### Documentation Changes Required
yes

### CODE REVIEW

**Comment by chmst** [comment]
I would like to have another icon for "get a permission report" - but not sure what could be a good visualisation. Maybe the eye?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22190">issues.joomla.org/tracker/joomla-cms/22190</a>.</sub>

**Comment by brianteeman** [comment]
I wouldn't use the eye as we use that for revealing a password - I actually thought the icon you chose was perfect. I can't think of another really suitable one. 

I really like this PR

**Comment by chmst** [comment]
Thank you. So I let this icon. It is also a a11y improvement for people with motoric disabilities.

**Comment by brianteeman** [comment]
Off topic. In the UK the word spastic although technically correct has become a very offensive word and we don't use it.

**Comment by chmst** [comment]
Thank you @brianteeman. How embarassing. It was not my intention

**Comment by brianteeman** [comment]
i was sure it wasn't

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 60914a43fa69707ab63c83695beef18c2cd7171f<br /><br />Have tested patch successfully. OT: As my patchtester doesn't work at the moment, I changed the file: administrator/components/com_users/tmpl/groups/default.php accordingly. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22190">issues.joomla.org/tracker/joomla-cms/22190</a>.</sub>

**Comment by brianteeman** [comment]
What do you think about removing the word "advanced" from the heading. Its a pretty long heading in english and possibly even longer in other languages

**Comment by ChristineWk** [comment]
OT: As my patchtester doesn't work at the moment, I changed the file: administrator/components/com_users/tmpl/groups/default.php accordingly.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22190">issues.joomla.org/tracker/joomla-cms/22190</a>.</sub>

**Comment by chmst** [comment]
I'd like to remove the "advanecd". It has not information, because no "not advanced" report exists. 
Can you do it?

**Comment by brianteeman** [comment]
just add the change to this pr

**Review by brianteeman — APPROVED** [review]
approve language changes

**Comment by chmst** [comment]
The "Advanced" is still in some Texts. Not sure what it exactly means, so I did not change it for other language keys.

**Comment by brianteeman** [comment]
Will you be making a similar change to the users view?

**Comment by chmst** [comment]
The change itself is easy and I will do it. Let's see ifthere is sufficient place in the table head

**Comment by brianteeman** [comment]
we might be able to just use "permissions" in the head

**Comment by chmst** [comment]
I've lost my struggle with github, wanted to make a new PR but now it is here

**Review by brianteeman — CHANGES_REQUESTED** [review]
Make the header text the same on both for consistency. Permissions on its own looks fine to me

**Comment by brianteeman** [comment]
And what to do you think about removing the ability to conditionally display this column. I only did it that way in j3 to avoid any discussion about it being a b/c change. Personally I would remove the condition but I'm ok if you want to keep it

**Comment by chmst** [comment]
I agree to remove it. Who can see users and usergroups can also see the permissions. I did not know the reason for the condition. When I leaned 1.6.. 2.5, the permissions were only visible in debug mode, maybe for performance?

**Comment by brianteeman** [comment]
Iirc they were an afterthought. Let's remove them for J4 and move forward. Thanks for doing this.

**Comment by chmst** [comment]
There are params in the "Advanced Tab".  Kind of "hidden secrets" :)

File: administrator/components/com_users/tmpl/groups/default.php [hunk 1]
```diff
@@ -50,6 +50,11 @@
 								<th scope="col">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_USERS_HEADING_GROUP_TITLE', 'a.title', $listDirn, $listOrder); ?>
 								</th>
+								<?php if ($debugGroups) : ?>
+									<th style="width:10%" class="text-center">
```
Comment (Quy) [inline administrator/components/com_users/tmpl/groups/default.php]: Add `scope="col"`

File: administrator/components/com_users/tmpl/groups/default.php [hunk 2]
```diff
@@ -50,6 +50,11 @@
...
+										<?php echo Text::_('COM_USERS_DEBUG_GROUP'); ?>
+									</th>
+								<?php endif;?>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/groups/default.php]: Add space after `;`

File: administrator/components/com_users/tmpl/users/default.php [hunk 3]
```diff
@@ -48,6 +48,11 @@
 								<th scope="col">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_USERS_HEADING_NAME', 'a.name', $listDirn, $listOrder); ?>
 								</th>
+								<?php if ($debugUsers) : ?>
+									<th sclass="text-center">
+										<?php echo Text::_('COM_USERS_DEBUG_USER'); ?>
+									</th>
+								<?php endif;?>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/users/default.php]: Add space after `;`

File: administrator/components/com_users/tmpl/users/default.php [hunk 4]
```diff
@@ -48,6 +48,11 @@
 								<th scope="col">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_USERS_HEADING_NAME', 'a.name', $listDirn, $listOrder); ?>
 								</th>
+								<?php if ($debugUsers) : ?>
+									<th sclass="text-center">
```
Comment (Quy) [inline administrator/components/com_users/tmpl/users/default.php]: Add `scope="col"`

**Comment by brianteeman** [comment]
Yes. it was me that put them there. They can go now as well.

**Comment by brianteeman** [comment]
Now that we have changed the title to just "Permissions" can you update the following as well please for consistency
```
COM_USERS_VIEW_DEBUG_GROUP_TITLE="Advanced Permissions Report for Group #%d, %s"
COM_USERS_VIEW_DEBUG_USER_TITLE="Advanced Permissions Report for User #%d, %s"
```

Then it's all good to go for me.

**Review by brianteeman — APPROVED** [review]
approve language changes

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 4cbbe993128c98858c8c5adf334abd5622d4d32d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22190">issues.joomla.org/tracker/joomla-cms/22190</a>.</sub>

**Comment by brianteeman** [comment]
thanks @chmst great improvement

**Comment by alikon** [comment]
yes indeed
can be backported on 3.X too ? @chmst

**Comment by brianteeman** [comment]
@alikon if backported it would need to keep the conditional display of the column

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 4cbbe993128c98858c8c5adf334abd5622d4d32d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22190">issues.joomla.org/tracker/joomla-cms/22190</a>.</sub>

**Comment by alikon** [comment]
@brianteeman  even if

**Comment by chmst** [comment]
I'll do it ;) because every times when I'm in the userr administration I click the wrong link

**Comment by chmst** [comment]
@brianteeman @alikon what do you think? for staging or for 3.9?

**Comment by brianteeman** [comment]
staging

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22190">issues.joomla.org/tracker/joomla-cms/22190</a>.</sub>

**Comment by laoneo** [comment]
Thanks

**Comment by PhilETaylor** [comment]
While this is a **million times better than before,** Im not convinced its the right place for it. 

I think the notes and the APR should be tabs in the Edit User view instead as thats the more logical place for them

The Add Notes is an action, so could also have been in the actions menu button. (which I dont see in these screenshots, darn admin template strikes again)

**Comment by chmst** [comment]
I think it is a great idea to have the permissions also in a tab in the edit screen.

But when I am working with ACL, I use to check the permissions of users one by one in the list view and compare if it is right. It would cost me more time to go to the edit screen for every user. So I think it is all right to have the links on the list view.

**Comment by brianteeman** [comment]
moving add a note to the toolbar is a good idea

**Pull request merged**